### ISSUE TITLE
Side navigation toggle button doesn't work on `space`

### ISSUE THREAD

**Issue by bartaz** [canonical/vanilla-framework#4559]
via @petermakowski comment https://github.com/canonical/juju.is/pull/431#issuecomment-1231665391

> `.p-side-navigation__toggle` element currently behaves like a link (`space` key does not trigger it on press, but that's just one example). Seems that a button element/role might be better suited here. 

**Describe the bug**

The toggle button for side navigation component is a link, not a button, and because of that you can't click it using `space` key (works only with `enter`).

It looks like a button and ideally should behave like one.
